core/templates/components/question-directives/questions-list/questions-list.component.ts

```diff
@@ -273,51 +273,69 @@ export class QuestionsListComponent implements OnInit, OnDestroy {
-      return;
+      questionDeletionIsAllowed = false;
+      return questionDeletionIsAllowed;
     }
 
-    this.deletedQuestionIds.push(questionId);
-    // For the case when, it is in the skill editor.
-    if (this.allSkillSummaries.length === 0) {
-      this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
-        questionId, [
-          {
-            id: this.selectedSkillId,
-            task: 'remove'
-          } as SkillLinkageModificationsArray
-        ]
-      ).then(() => {
-        this.questionsListService.resetPageNumber();
-        this.questionsListService.getQuestionSummariesAsync(
-          this.selectedSkillId, true, true);
-        this.alertsService.addSuccessMessage('Deleted Question');
-        this._removeArrayElement(questionId);
-      });
-    } else {
-      this.allSkillSummaries.forEach((summary) => {
-        if (summary.getDescription() === skillDescription) {
-          this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
-            questionId, [
-              {
-                id: summary.getId(),
-                task: 'remove'
-              } as SkillLinkageModificationsArray
-            ]
-          ).then(() => {
-            this.questionsListService.resetPageNumber();
-            this.questionsListService.getQuestionSummariesAsync(
-              this.selectedSkillId, true, true);
-            this.alertsService.addSuccessMessage('Deleted Question');
-            this._removeArrayElement(questionId);
-          });
+    let numberOfQuestions = this.questionSummariesForOneSkill.length;
+    await this.skillBackendApiService
+      .getTopicNamesWithGivenSkillAssignedForDiagnosticTest(
+        this.selectedSkillId).then((topicNames) => {
+        if ((topicNames.length > 0) && (
+          numberOfQuestions <=
+            AppConstants.MINIMUM_QUESTION_COUNT_FOR_A_DIAGNOSTIC_TEST_SKILL)) {
+          let alertMessge = (
+            'This skill is used as the only diagnostic test skill in the ' +
+            'following topics: ' + topicNames.join(', ') + '. Please remove ' +
+            'this skill from the diagnostic tests of those topics first.'
+          );
+          this.alertsService.addInfoMessage(alertMessge, 7000);
+          questionDeletionIsAllowed = false;
         }
       });
-    }
+    return questionDeletionIsAllowed;
+  }
+
+  removeQuestionSkillLinkAsync(questionId: string, skillId: string): void {
+    this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
+      questionId, [
+        {
+          id: skillId,
+          task: 'remove'
+        } as SkillLinkageModificationsArray
+      ]
+    ).then(() => {
+      this.questionsListService.resetPageNumber();
+      this.questionsListService.getQuestionSummariesAsync(
+        this.selectedSkillId, true, true);
+      this.alertsService.addSuccessMessage('Deleted Question');
+      this._removeArrayElement(questionId);
+    });
+  }
+
+  deleteQuestionFromSkill(
```

Comment:
Ditto, I am not sure this is a "delete"?

Reply:
Updated the naming in every place to reflect remove instead of delete. Please take a look.